src/Generator/Generators/CSharp/CSharpSources.cs

```diff
@@ -2768,9 +2794,17 @@ into context
                 // Special case for indexer - needs to dereference if the internal
                 // function is a pointer type and the property is not.
                 if (retType.Type.IsAddress() &&
-                    retType.Type.GetPointee().Equals(returnType) &&
-                    returnType.IsPrimitiveType())
-                    WriteLine("return *{0};", marshal.Context.Return);
```

Comment:
I know this is existing code, but this is doing the job of the marshaler. We should move it to marshal if possible.